### PR TITLE
feat: add \emph

### LINKED ISSUE #3566: add \emph
`\text{a}` and `\textit{a}` compile find with KaTeX, `\emph{a}` doesn't. Perhaps that'd be a worthwhile addition.

### PR DESCRIPTION
**Pull request by grigoriy-reshetniak**
**What is the previous behavior before this PR?**
`\emph` wasn't supported

**What is the new behavior after this PR?**
`\emph` is supported

image

Implements default behavior of making text italic if it's not and making text upright if it's italic. Please let me know ff more complex rules are required.

Closes #3566

### CODE REVIEW

File: src/functions/text.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ const textFontWeights = {
 const textFontShapes = {
     "\\textit": "textit",
     "\\textup": "textup",
+    "\\emph": "textit",
```
Comment (edemaine) [inline src/functions/text.js:22]: I think it would be better to avoid including this as a shape (see below).

File: src/functions/text.js [hunk 2]
```diff
@@ -30,9 +31,11 @@ const optionsWithFont = (group, options) => {
         return options.withTextFontFamily(textFontFamilies[font]);
     } else if (textFontWeights[font]) {
         return options.withTextFontWeight(textFontWeights[font]);
-    } else {
-        return options.withTextFontShape(textFontShapes[font]);
+    } else if (options.fontShape === "textit" && font === "\\emph") {
```
Comment (edemaine) [inline src/functions/text.js:34]: I'd suggest changing this to `if (font === "\emph")` and using `options.withTextFontShape` (enforcing text mode) with `"textit"` or `"textup"` according to `options.fontShape`.

**Review by edemaine — CHANGES_REQUESTED** [review]
I didn't realize that `\emph` works in math mode.  Your implementation seems to exactly match LaTeX behavior:

```latex
x+\emph{foo \emph{bar}}+\textit{foo \emph{bar}}
```

image

LaTeX:
image

In addition to the implementation comments below, can you add some tests? Something like ``expect`\emph{foo \emph{bar}}`.toParseLike`\textit{foo \textup{bar}}` ``.

**Comment by grigoriy-reshetniak** [comment]
My bad, I should've added test from the beginning

**Review by edemaine — APPROVED** [review]
Looks good! I re-organized the Fonts docs table to include `\emph`. Could you double check it looks good before I merge?

**Review by grigoriy-reshetniak — COMMENTED** [review]
Looks good to me.

**Pull request merged**